### ISSUE TITLE
`LiteralString` should not be compatible with `str` in an invariant context

### ISSUE THREAD

**Issue by erictraut** [python/mypy#16169]
Consider the following code sample. 

```python
from typing import LiteralString

def func1(a: list[LiteralString]):
    v1: list[str] = a   # Should generate a type error but does not currently
```

I just fixed a [related bug in pyright](https://github.com/microsoft/pyright/issues/5999).

Mypy version 1.5.1

**Comment by JelleZijlstra** [comment]
This is because internally mypy currently treats LiteralString as an alias of str as a stopgap until we get full support (#12554). Closing as a duplicate of that issue.
